session.go

```diff
@@ -3143,23 +3146,26 @@ func (c *Collection) UpsertId(id interface{}, update interface{}) (info *ChangeI
 //
 //     http://www.mongodb.org/display/DOCS/Removing
 //
-func (c *Collection) Remove(selector interface{}) error {
+func (c *Collection) Remove(selector interface{}) (info *ChangeInfo, err error) {
 	if selector == nil {
 		selector = bson.D{}
 	}
 	lerr, err := c.writeOp(&deleteOp{c.FullName, selector, 1, 1}, true)
 	if err == nil && lerr != nil && lerr.N == 0 {
-		return ErrNotFound
+		return &ChangeInfo{}, ErrNotFound
 	}
-	return err
+	if err == nil && lerr != nil {
+		info = &ChangeInfo{Updated: lerr.modified, Matched: lerr.N}
```

Comment:
For the remove I think you need to set the ```Removed``` field instead.
https://docs.mongodb.com/manual/reference/method/db.collection.remove/#writeresult